src/components/EditWebhooks.js

```diff
@@ -0,0 +1,259 @@
+        if (!(x.webhookUrl in webhooks)) {
+          webhooks[x.webhookUrl] = pick(x, ['webhookUrl']);
+          webhooks[x.webhookUrl].activities = [];
+        }
+        webhooks[x.webhookUrl].activities.push(x.type);
+      });
+
+      return { ...state, webhooks: Object.values(webhooks) };
+    }
+
+    return state;
+  }
+
+  constructor(props) {
+    super(props);
+    const { intl } = props;
+
+    this.state = {
+      modified: false,
+      webhooks: {},
+      status: null,
+    };
+
+    this.messages = defineMessages({
+      'webhooks.url.label': {
+        id: 'webhooks.url.label',
+        defaultMessage: 'URL',
+      },
+      'webhooks.types.label': {
+        id: 'webhooks.types.label',
+        defaultMessage: 'Activities',
+      },
+      'webhooks.add': {
+        id: 'webhooks.add',
+        defaultMessage: 'add another webhook',
+      },
+      'webhooks.remove': {
+        id: 'webhooks.remove',
+        defaultMessage: 'remove webhook',
+      },
+      loading: { id: 'loading', defaultMessage: 'loading' },
+      save: { id: 'save', defaultMessage: 'save' },
+      saved: { id: 'saved', defaultMessage: 'saved' },
+    });
+
+    this.fields = [
+      {
+        name: 'webhookUrl',
+        maxLength: 255,
+        type: 'url',
+        label: intl.formatMessage(this.messages['webhooks.url.label']),
+        required: true,
+      },
+      {
+        name: 'activities',
+        type: 'select',
+        label: intl.formatMessage(this.messages['webhooks.types.label']),
+        options: events.sort().map(event => {
+          return { value: event, label: event };
+        }),
+        multiple: true,
+        defaultValue: [],
+        required: true,
+      },
+    ];
+  }
+
+  editWebhook = (index, fieldname, value) => {
+    const { webhooks } = this.state;
+    webhooks[index][fieldname] = value;
+    this.setState({ webhooks, modified: true });
+  };
+
+  addWebhook = webhook => {
+    const { webhooks } = this.state;
+    webhooks.push(webhook || {});
+    this.setState({ webhooks, modified: true });
+  };
+
+  removeWebhook = index => {
+    const { webhooks } = this.state;
+    if (index < 0 || index > webhooks.length) return;
+    webhooks.splice(index, 1);
+    this.setState({ webhooks, modified: true });
+  };
+
+  handleSubmit = async () => {
+    this.setState({ status: 'loading' });
+    const { webhooks } = this.state;
+    const notifications = [];
+    for (const notification of webhooks) {
+      if (!notification.webhookUrl) continue;
+
+      for (const activity of notification.activities) {
+        if (!activity) continue;
+
+        notifications.push({
+          id: notification.id,
+          channel: 'webhook',
+          type: activity,
+          active: true,
+          webhookUrl: notification.webhookUrl,
+        });
+      }
+    }
+
+    await this.props.editNotifications({ id: this.props.data.Collective.id, notifications });
```

Comment:
What happens if the mutation fails here?

Reply:
From what I've tested, errors are not surfaced on the frontend. We should display them.

To test errors, you can add a URL like `http://localhost:8888` (API refuse it)

Reply:
It now displays an error on the page.

![image](https://user-images.githubusercontent.com/11166933/57794618-f3e29780-773b-11e9-8328-8c69ddeaa221.png)